File: NimGame/Count_and_Say/countAndSay.java
class Solution {
    public String countAndSay(int n) {
        String str = "1";
        //if (n < 2) throw new IllegalArgumentException("Error");
        
        while (n > 1) {
            int count = 0;
            char c = str.charAt(0);
            StringBuilder pre = new StringBuilder();
            for (int i = 0; i <= str.length(); i++) {
                if (i != str.length() && str.charAt(i) == c) {
                    count++;
                } else {
                    pre.append(count);
                    pre.append(c); //str.charAt(i - 1));
                    if (i != str.length()) {
                        c = str.charAt(i);
                        count = 1;
                    }
                }
            }
            str = pre.toString();
            n--;
        }
        return str;
    }
}

File: Count_and_Say/countAndSay.cpp
class Solution {
public:
    string countAndSay(int n) {
        string str = "1";

        while (n > 1) {
            int count = 0;
            string pre = "";
            char c = str[0];
            for (int i = 0; i <= str.size(); i++) {
                if (str[i] == c) {
                    count++;
                } else {
                    pre += to_string(count) + c;
                    c = str[i];
                    count = 1;
                }
            }
            str = pre;
            n--;
        }
        return str;
    }
};
